navigation/libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationMetricsWrapper.java

```diff
@@ -0,0 +1,66 @@
+package com.mapbox.services.android.navigation.v5.navigation;
+
+import android.location.Location;
+
+import com.mapbox.services.android.navigation.BuildConfig;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+import com.mapbox.services.android.telemetry.MapboxTelemetry;
+import com.mapbox.services.android.telemetry.navigation.MapboxNavigationEvent;
+
+import java.util.Date;
+
+class NavigationMetricsWrapper {
+
+  static void arriveEvent(SessionState sessionState, RouteProgress routeProgress, Location location) {
+    MapboxTelemetry.getInstance().pushEvent(MapboxNavigationEvent.buildArriveEvent(
+      "mapbox-navigation-android", BuildConfig.MAPBOX_NAVIGATION_VERSION_NAME,
+      sessionState.sessionIdentifier(), location.getLatitude(), location.getLongitude(),
+      routeProgress.directionsRoute().getGeometry(), "",
+      (int) routeProgress.directionsRoute().getDistance(),
+      (int) routeProgress.directionsRoute().getDuration(),
+      sessionState.rerouteCount(), sessionState.startTimestamp(), (int) routeProgress.distanceTraveled(),
+      sessionState.mockLocation(),"null", "null",
+      sessionState.originalGeometry(), sessionState.originalDistance(),
+      sessionState.originalDuration(), "null", sessionState.currentStepCount(),
+      sessionState.originalStepCount()
+    ));
+  }
+
+  static void cancelEvent(SessionState sessionState, RouteProgress routeProgress, Location location) {
+    MapboxTelemetry.getInstance().pushEvent(MapboxNavigationEvent.buildCancelEvent(
+      BuildConfig.APPLICATION_ID, BuildConfig.MAPBOX_NAVIGATION_VERSION_NAME, sessionState.sessionIdentifier(),
+      location.getLatitude(), location.getLongitude(),
+      routeProgress.directionsRoute().getGeometry(), "",
+      (int) routeProgress.directionsRoute().getDistance(),
+      (int) routeProgress.directionsRoute().getDuration(),
+      sessionState.rerouteCount(), sessionState.startTimestamp(), (int) routeProgress.distanceTraveled(),
+      (int) routeProgress.distanceRemaining(), (int) routeProgress.durationRemaining(),
+      sessionState.mockLocation(),
+      "null", "null", sessionState.originalGeometry(),
+      sessionState.originalDistance(), sessionState.originalDuration(), "null",
+      new Date(), sessionState.currentStepCount(), sessionState.originalStepCount()
```

Comment:
This seems to always be setting the arrival timestamp to now.

Reply:
ah my bad, fixing. Note that for now, I'm using `"null"` for now until I fix the issue downstream